configuration.nix

```diff
@@ -5,22 +5,20 @@
 { config, pkgs, ... }:
 
 let
-  posh = global_args: run_args: image: (pkgs.writeScriptBin "posh" ''
-      #! ${pkgs.bash}/bin/bash
-      source /etc/profile
-      tty -s && tty="-t" || quiet="-q"
-      test -S "$SSH_AUTH_SOCK" && ssh="-v $SSH_AUTH_SOCK:$SSH_AUTH_SOCK -e SSH_AUTH_SOCK"
-      ${pkgs.podman}/bin/podman pull $quiet ${image} >/dev/null
-      shift
-      exec ${pkgs.podman}/bin/podman ${global_args} run --rm -i $tty $ssh -v ~/:/root -w /root --network host ${run_args} ${image} $@
-    '')
-    .overrideAttrs(attrs: attrs // {
-        passthru = {
-          shellPath = "/bin/posh";
-        };
-    });
+  steveej_repo = let
+      rev = "72f2fae5e903b82e3b2f55a49b54428158945fea";
+    in (import
+      (builtins.fetchTarball {
+        url = "https://gitlab.com/steveeJ/infra/-/archive/${rev}/infra-${rev}.tar.bz2";
+        sha256 = "016138k4ri88k11iydgi8r9q5zdps9bylza9c1m9r8867zms0c2c";
+      })
+    ) { inherit pkgs; };
```

Comment:
Is there an option here besides blindly fetching a tarball? A git URI would be better...

Reply:
Using `builtins.fetchgit` or something similar from `pkgs` is possible and I can switch to that if you prefer it.

Reply:
Yeah. Tarballs are very opaque. I don't know what I'm sucking in.